// authenik8-core/dist/controllers/authController.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.authController = void 0;
exports.authController = {
    register: (options) => async (req, res) => {
        try {
            if (!options.registerHandler) {
                return res.status(500).json({
                    error: "No registerHandler provided",
                });
            }
            const user = await options.registerHandler(req);
            res.json({ success: true, user });
        }
        catch (err) {
            res.status(400).json({ error: err.message });
        }
    },
    login: (options) => async (req, res) => {
        try {
            if (!options.loginHandler) {
                return res.status(500).json({
                    error: "No loginHandler provided",
                });
            }
            const user = await options.loginHandler(req);
            const accessToken = options.instance.signToken(user);
            const refreshToken = await options.instance.generateRefreshToken(user);
            res.json({ accessToken, refreshToken });
        }
        catch (err) {
            res.status(401).json({ error: err.message });
        }
    },
    refresh: (options) => async (req, res) => {
        try {
            const { refreshToken } = req.body;
            const tokens = await options.instance.refreshToken(refreshToken);
            res.json(tokens);
        }
        catch (err) {
            res.status(403).json({ error: err.message });
        }
    },
    logout: (options) => async (req, res) => {
        try {
            // If you later add revoke support
            res.json({ success: true });
        }
        catch (err) {
            res.status(400).json({ error: err.message });
        }
    },
    Admin: (options) => async (req, res) => {
        var _a;
        try {
            const token = (_a = req.headers.authorization) === null || _a === void 0 ? void 0 : _a.split(" ")[1];
            if (!token) {
                return res.status(401).json({ error: "Missing token" });
            }
            const user = await options.instance.verifyToken(token);
            res.json(user);
        }
        catch {
            res.status(401).json({ error: "Unauthorized" });
        }
    },
};
